LLVM.Imports.pas

```diff
@@ -29,7 +30,26 @@ implementation
 
 function TLLVMRef.IsValid: Boolean;
 begin
-  Result := Assigned(Value);
+
```

Comment:
any particular reason for this?
IsValid is supposed to work like Assigned() and therefore simply returning true if the value is not nil, without validating memory.

Reply:
problem check if a pointer is valid, the pointer might have a value but not be valid and therefore assigned returns true instead of false.
a better control, even if not definitive, is to check if the location does not return an error while reading and writing it

Reply:
If a pointer has a valude but is not valid, you have an entirely different issue. 
Your code might even execute properly on an ivalid pointer which has a value.

Reply:
in many cases I have had problems of this type,
which then is the same problem as the IsbadReadptr type function of windows, assigned only checks that the value is not null.
Anyway you can also discard the change there are no problems.

https://stackoverflow.com/questions/496034/most-efficient-replacement-for-isbadreadptr